scalding-core/src/main/scala/com/twitter/scalding/serialization/WrappedSerialization.scala

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2014 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package com.twitter.scalding.serialization
+
+import org.apache.hadoop.io.serializer.{ Serialization => HSerialization, Deserializer, Serializer }
+import org.apache.hadoop.conf.{ Configurable, Configuration }
+
+import java.io.{ DataInputStream, DataOutputStream, InputStream, OutputStream }
+import com.twitter.bijection.{ Injection, JavaSerializationInjection, Base64String }
+import scala.collection.JavaConverters._
+
+/**
+ * WrappedSerialization wraps a value in a wrapper class that
+ * has an associated Binary that is used to deserialize
+ * items wrapped in the wrapper
+ */
+class WrappedSerialization[T] extends HSerialization[T] with Configurable {
+
+  import WrappedSerialization.ClassSerialization
+
+  private var conf: Option[Configuration] = None
+  private var serializations: Iterable[ClassSerialization[_]] = Nil
+
+  override def getConf: Configuration = conf.get
+  override def setConf(config: Configuration) {
+    conf = Some(config)
+    serializations = WrappedSerialization.getBinary(config)
+  }
+
+  def accept(c: Class[_]): Boolean =
+    serializations.exists { case (cls, _) => cls == c }
+
+  def getSerialization(c: Class[T]): Option[Serialization[T]] =
+    serializations.collectFirst { case (cls, b) if cls == c => b }
+      // This cast should never fail since we matched the class
+      .asInstanceOf[Option[Serialization[T]]]
+
+  def getSerializer(c: Class[T]): Serializer[T] =
+    new BinarySerializer(getSerialization(c).getOrElse(sys.error(s"Class: ${c} not found")))
+
+  def getDeserializer(c: Class[T]): Deserializer[T] =
+    new BinaryDeserializer(getSerialization(c).getOrElse(sys.error(s"Class: ${c} not found")))
+
+}
+
+class BinarySerializer[T](buf: Serialization[T]) extends Serializer[T] {
+  private var out: OutputStream = _
+  def open(os: OutputStream): Unit = {
+    out = os
+  }
+  def close(): Unit = { out = null }
```

Comment:
seems like, but we're not 100% sure the contract is well defined between Hadoop and what Cascading does. Want to check into that?